### PR TITLE
07 vote on review

### PR DESCRIPTION
**Pull request by StaceyCP**
Voting on a review is complete with error handling 👍

### CODE REVIEW

File: src/components/ReviewPage.jsx [hunk 1]
```diff
@@ -1,22 +1,43 @@
...
     }, [review_id])
-    const configuredDate = new Date(review.created_at).toDateString()
-    if (!reviewLoading) {
+    
+    let vote = {
+        inc_votes: 1
+    }
+ 
+    const handleVote = (num) => {
+        vote.inc_votes = num
```
Comment (NRMSMITH) [inline src/components/ReviewPage.jsx:34]: Ideally, you should change the state before the api call, as that way you are optimistically rendering the desired result. The catch block below is great too.

**Review by NRMSMITH — COMMENTED** [review]
can you rethink your voting system - love/Like/dislike is very arbitrary and not intuitive for the user. Buttons that read the amount of votes being given might be more intuitive for a screen reader or someone who is ESL. 

Also check comment about optimistic rendering - we should be able to see the vote increase before the error notice(which you might want to make a simple sentence on the page rather than changing the entire page).

File: src/components/ReviewPage.jsx [hunk 2]
```diff
@@ -62,24 +66,27 @@ function ReviewPage() {
...
+                        <button className="reaction-page_reaction dislike" type="button" aria-label="Remove 1 from the review likes" onClick={() => {
+                            if (!reacted) {
+                                handleVote(-1);
+                                setReacted(true);
+                                setShowModal(true);
+                            }}}></button>
+                        <p>{voteCount} likes</p>
+                        {reacted && showModal && <Modal text={"Whoops! Sorry, you can only react once"} setShowModal={setShowModal}/>}
                     </div>
                     <hr></hr>
```
Comment (MKD01) [inline src/components/ReviewPage.jsx]: I really like the idea behind this, but it doesnt seem to work the way you may think.

Clicking the like or dislike buttons the first time shows an error modal pop up, which i dont think should be the case.

After clicking the like or dislike the opposite can be clicked but does not change the vote amount.

If you click one of the buttons and then click else where on the page the button that was selected is no longer obvious, but that is just something to consider at this point.

**Review by MKD01 — CHANGES_REQUESTED** [review]
Great work over all but please do make those changes first

**Review by MKD01 — APPROVED** [review]
Great work so far, but once a user has voted they cannot remove their vote or vote the opposite way.

**Pull request merged**